### ISSUE TITLE
Problemas Detectados

### ISSUE THREAD

**Issue by UndMe** [TecnologiaVideojuegos/proyecto-videojuego-team6#23]
Creo que seria bueno tener un Issue de los problemas que podamos ir detectando a la hora de testear el juego. 
Como he comentado en uno que tenemos de ESTADO, a la hora de generar la cuenta y darle al boton de atras, se sale del juego.
Otra cosa que he visto, es que las cuentas se generan como de forma "temporal", osea cuando se sale del juego no se mantiene, tienes que volver a crearla.

**Comment by JuanCasado** [comment]
Ya he corregido lo de que se cerraba, estaba leyendo el botón de la pantalla anterior y no el de la actual

**Comment by JuanCasado** [comment]
Las cuentas se generan de forma permanente pero tienes que salir del juego con los botones del juego, no con el de la pantalla

**Comment by JuanCasado** [comment]
Si me decís un nombre y una contraseña (inventados) os meto una cuenta de desarrollador cada uno, con las cuentas de desarrollador puede cambiar de forma activa el dinero u otros stats o ver datos del juego desde la terminal

**Comment by JuanCasado** [comment]
En cualquier caso en la terminal te dice si se cargan bien los datos o no, sale CREATE BASE DATA cuando se crean datos por primera vez, DATA LOADED si se cargaron datos antiguos y DATA SAVED cada vez que se guardan datos nuevos de forma correcta @UndMe

**Comment by UndMe** [comment]
Vale, voy a mirar ahora si me deja entrar. Me imaginaba que estaba como leyendo la X de salir en lugar del boton de volver.
Respecto al tema de la terminal, si puedes decirme porfavor como sacarla en netbeans, porque no tengo ni idea....

Y el tema de las cuentas me parece bien,
Usuario: UndMe
Contraseña: qwerty

**Comment by JuanCasado** [comment]
La terminal es donde pone output, creo que sale abajo por defecto

**Comment by UndMe** [comment]
Es cierto, sale todo correcto con DATA LOADED, la verdad que esta bastante chulo hasta el momento todo lo que llevamos.
Como digo, ami el tema de la musica todavia no me suena, y la X de salir (la que esta en el menu) cuando le doy se me queda en: programa no responde, mirar a ver si os funciona bien el boton a vosotros.

**Comment by JuanCasado** [comment]
Los comandos de desarrollador son:
SCENES                    -> muestra las escenas activas
DEBUG  0 | 1            ->  habilita o quita el modo debug (los rectángulos grises del juego)
BULLETS N              ->  te pone las balas indicadas
HEALTH N                ->  te pone la vida indicada
STAGE N                  ->  te permite entrar hasta el nivel indicado (hay 10)
GUN N M                 ->  le da al arma N (del 0 al 4) el nivel M (del 0 al 5)
SCOPE this              ->  te dice tus stats

**Comment by JuanCasado** [comment]
La música es un poco rara como la maneja slick, en el mac me va en los altavoces y los cascos normales, pero en los bluetooth no, prueba a ponerle unos cascos, configurarlos para que salga todo el sonido por ello y luego arranca, es lo que tuve que hacer para que sonara en los míos, en cuanto a lo del botón lo miro, pero a mi me cierra correctamente.

**Comment by UndMe** [comment]
Cierto, los comandos funcionan sin problema. Respecto al tema de la musica lo he probado con cascos que tienen jack y con otros que van por usb y no me suena tampoco... yo creo que es mas por como maneja windows la musica o algo, seguramente si lo pruebo en mac funcione. Luego cuando llegue a casa lo puedo probar y asi salimos de la duda.

Lo del boton de salir me refiero una vez has jugado una partida por lo menos, porque si que lo cierra bien en cuanto entras, pero despues de jugar no.

**Comment by JuanCasado** [comment]
A mi me cierra sin problemas en todo momento y no hay razón aparente para que no lo hiciera, la librería de sonido que es OpenAL debe cerrarse de forma especial y no le vale con el System.exit(0), he cambiado el modo de cierre para adaptarlo a la librería aunque no debería afectar porque los buffers los vacía el OS, ahora lo hago a mano y a lo mejor eso ayuda, si siguieran existiendo errores por favor sube un vídeo u di algo más detallado del error porque con lo que me dices no sé identificar lo que pasa :'( @UndMe

**Comment by UndMe** [comment]
Sigue sin sonarme todavia.... como te digo luego lo probare en el mac que tengo en casa a ver si eso fuese el problema y te comento. Puede que sea por el formato de la cancion? que necesite algun formato especial el slick ?

**Comment by JuanCasado** [comment]
Las canciones están en .ogg que es uno de los que entiende slick, alguna está en wav que también le sirve, pero las librería de OpenAL debería ser la que maneje el archivo y no debería fallar

**Comment by UndMe** [comment]
@JuanCasado confirmado, el problema esta en windows. Porque estoy probándolo ahora mismo en el MAC y la música funciona sin problema alguno, asique nose porque motivo sera....

Otra cosa que me he fijado, en la pantalla del mac se ve bastante bien, sin embargo en windows se corta un poco por abajo, creo que habría que mirar el tema ese de la resolución, o poner incluso (que nose si se podra), una opción en el menu para dar al usuario a elegir.

**Comment by JuanCasado** [comment]
Pasa una foto de como se ve en windows porfa, debería adaptarse al tamaño de la pantalla
@UndMe

**Comment by UndMe** [comment]
![captura](https://user-images.githubusercontent.com/37188090/39973064-e9b4791e-571a-11e8-8d93-f739a25a249d.png)

Mal como tal no se ve, pero al tener arriba la barra con el nombre del juego y demas, lo corta un poco. Si desbloqueo la barra de inicio si se veria bien, pero creo que normalmente la gente suele tenerla bloqueada.
Por cierto, el tema de la pantalla completa habria que verlo tambien no?

**Comment by JuanCasado** [comment]
Bájate la versión que voy a subir ahora que tiene la pantalla completa y lo de windows arreglado @UndMe

**Comment by JuanCasado** [comment]
Está ya subida, por cierto, para salir de dentro de una partida hay que darle a escape o a borrar (<-) entonces entras en el menú de pausa y te deja salir

**Comment by UndMe** [comment]
:O esta increible ahora, me gusta asi si!! ahora se ve como dios manda !!
Vale, pues entonces el unico problema que queda como antes te habia comentado es a la hora de salir, en mac no me ha pasado pero en windows si. 
Cuando terminas de jugar y sales al menu principal, al darle a la X para salir se queda colgado. Ya te digo tienes que jugar por lo menos una partida porque sino si que funciona bien.

**Comment by UndMe** [comment]
![captura](https://user-images.githubusercontent.com/37188090/39973142-5a678fb0-571c-11e8-82da-fa2b3a99f925.jpg)
Este es el error que me da al cerrarse de golpe, por si te sirve en algo.

**Comment by JuanCasado** [comment]
Que es lo que pone a la derecha del todo?

**Comment by JuanCasado** [comment]
Y arriba queda también un cacho del error jajajaja

**Comment by UndMe** [comment]
C:\Users\Jorge\Desktop\GIT\proyecto-videojuego-team6\Programacion\ShutterEarth\nbproject\build-impl.xml:1040: The following error occurred while executing this line:

C:\Users\Jorge\Desktop\GIT\proyecto-videojuego-team6\Programacion\ShutterEarth\nbproject\build-impl.xml:805: Java returned: -805306369

BUILD FAILED (total time: 1 minute 6 seconds)

Eso es exactamente lo que pone @JuanCasado

**Comment by JuanCasado** [comment]
Es un fallo interno del precompilador de net beans :/ 

Has probado a darle al simbolito que es un martillo y una escoba que está a la izquierda del PLAY verde de la barra superior? @UndMe

**Comment by UndMe** [comment]
Le acabo de dar y sale esto en rojo:

![imagen](https://user-images.githubusercontent.com/37188090/39973251-fedf24d0-571d-11e8-9e80-6be45c3d544a.png)

Tambien que prueben @Zaedlen @pablopg99 a ver si les pasa o si es solo mio, aunque como te digo en mac no me ha pasado antes, osea el problema lo esta dando siempre con windows... como la musica.

**Comment by JuanCasado** [comment]
Nah, pero eso son warnings de netBeans, eso no importa, puede que sea del java no actualizado o que lo mismo alguna de las librerías tiene una versión más moderna que slick todavía no soporta, eso no pasa nada

**Comment by UndMe** [comment]
@JuanCasado estoy probando el juego pero al darle para saltar se me sale de la pantalla el personaje y luego los bichos, salen disparados hacia arriba.

**Comment by JuanCasado** [comment]
Te pasaba ayer? @UndMe

**Comment by JuanCasado** [comment]
Prueba a ejecutar ahora, te va a salir por la terminal una serie de datos, me pasa una captura de ellos porfa, hay que ajustar las constantes del juego a que se adapten a las distintas pantallas

@UndMe

**Comment by UndMe** [comment]
@JuanCasado efectivamente fue ayer, pero ahora mismo que he actualizado todo lo nuevo que habeis subido a github no me pasa, ahora mismo funciona perfecto.
Ya veo la generacion de las habitaciones y todo, lo unico que cuando saltas no se queda en la habitacion que tenga encima, me imagino que eso todavia no estais programando.

**Comment by UndMe** [comment]
@JuanCasado no es un problema, pero creo que seria bueno a la hora de iniciar sesion que al darle al intro acceda a la cuenta, es decir, no usar el raton.... nose si slick lo permitira hacer.
Por otro lado como te comente al principio (solo el windows), la X de cerrar el juego me lo cuelga una vez he jugado alguna partida.

Y otra cosilla, que veo que alomejor seria bueno, es dejar que el personaje suba/baje por toda la planta y no unicamente por las escaleras, porque sino rapidamente el bicho va a quitarte las vidas. Nose como lo vereis vosotros, unicamente es una observacion.

**Comment by JuanCasado** [comment]
Haz un pull y prueba a crear una cuenta de usuario y a ver si crashea, creo que lo he parcheado para esas @UndMe

**Comment by JuanCasado** [comment]
Lo del enter se puede sin problemas, ahora lo meto y lo de las escaleras lo estuvimos pensando pero tay y como está hecho es bastante jodido cambiarlo ya así que yo lo dejaría como está, la forma de jugar no es entrar en la sala y disparar si no sin entrar disparar y caer, así no les da tiempo al golpearte

@UndMe

**Comment by UndMe** [comment]
Vale perfecto @JuanCasado, ahora con una nueva cuenta si funciona perfecto sin pillarse ni nada. Entonces para las cuentas de administradores siempre se me va a quedar pillado no?
Respecto a la mecanica, me parece bien. La verdad que no habia caido en directamente saltar y disparar, voy a probarlo.
